{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "44517616-daf0-430f-940f-76057cd27009",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from plotly import graph_objects as go\n",
    "import numpy as np\n",
    "from Bio import Phylo\n",
    "import plotly.io as pio\n",
    "import shutil\n",
    "import importlib\n",
    "from tqdm import tqdm\n",
    "from plotly.subplots import make_subplots\n",
    "from pathlib import Path\n",
    "if not Path('jw_utils').exists():\n",
    "    !git clone https://github.com/JonWinkelman/jw_utils.git\n",
    "from jw_utils import plotly_preferences as pprefs\n",
    "from jw_utils import itol_annotations as ia\n",
    "from jw_utils import ncbi_utils as nu\n",
    "from jw_utils import parse_fasta as pfa\n",
    "from jw_utils import hmmer_utils as hu\n",
    "from jw_utils import parse_gff as pgf\n",
    "from jw_utils import alignment_utils2 as au\n",
    "\n",
    "if not Path('orthofinder_utils').exists():\n",
    "    !git clone https://github.com/JonWinkelman/orthofinder_utils.git\n",
    "from orthofinder_utils import dash_ortho_parser_GTDB as dop\n",
    "from orthofinder_utils import run_orthofinder as ru\n",
    "import external_functions as ef\n",
    "from jw_utils import make_bac120_tree as mbt\n",
    "from ete3 import Tree"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c213228a-e362-418c-9ba0-06e31813d4e4",
   "metadata": {},
   "source": [
    "### Download proteomes and GFF files from NCBI datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ea71b341-502f-4384-9ad0-3cfdf8ec0ed3",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_genome_assemblies = {\n",
    "    'GCF_001077675.1': 'Acinetobacter baumannii ATCC 17978-mff',\n",
    "    'GCF_000018445.1': 'Acinetobacter baumannii ACICU',\n",
    "    'GCF_000021245.2': 'Acinetobacter baumannii AB0057',\n",
    "    'GCF_000770605.1': 'Acinetobacter baumannii AB5075',\n",
    "    'GCF_005281455.1': 'Acinetobacter nosocomialis M2',\n",
    "    'GCF_002055515.1': 'Acinetobacter calcoaceticus CA16',\n",
    "    'GCF_001682515.1': 'Acinetobacter gyllenbergii FMP01',\n",
    "    'GCF_000413935.1': 'Acinetobacter colistiniresistens NIPH 2036',\n",
    "    'GCF_000046845.1': 'Acinetobacter baylyi ADP1',\n",
    "    'GCF_000368805.1': 'Acinetobacter johnsonii ANC 3681',\n",
    "    'GCF_004208515.1': 'Acinetobacter halotolerans JCM 31009',\n",
    "    'GCF_001105265.1': 'Yersinia enterocolitica SC9312-78',\n",
    "    }\n",
    "figure_accs = list(figure_genome_assemblies.keys())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d6d3d09-9b58-4942-b4cc-00347d607d25",
   "metadata": {},
   "source": [
    "#### Download assembly summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "1df0f2e4-3173-4a9e-8255-f969199c6b69",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "temp_file created at /var/folders/vw/7lg51dfd3ql9g_j55xz_3dqh0000gn/T/tmpcpc52s_0\n"
     ]
    }
   ],
   "source": [
    "nu.download_assembly_summaries_from_list(figure_accs)\n",
    "summary_df = nu.make_summary_df('summaries.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "530317c0-fa99-4fa7-ae9f-89cbdd34b86b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCF_000046845.1.faa',\n",
       " 'GCF_002055515.1.faa',\n",
       " 'GCF_000021245.2.faa',\n",
       " 'GCF_000770605.1.faa',\n",
       " 'GCF_001077675.1.faa',\n",
       " 'GCF_000018445.1.faa',\n",
       " 'GCF_000413935.1.faa',\n",
       " 'GCF_001105265.1.faa',\n",
       " 'GCF_000368805.1.faa',\n",
       " 'GCF_005281455.1.faa',\n",
       " 'GCF_004208515.1.faa',\n",
       " 'GCF_001682515.1.faa']"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#nu.download_genomes_from_acclist(figure_accs)\n",
    "proteomes_dir = Path('./results/1_orthofinder_analysis/data/proteomes')\n",
    "proteomes_dir.mkdir(exist_ok=True, parents=True)\n",
    "nu.copy_ncbi_files('./ncbi_dataset/ncbi_dataset/data/', proteomes_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85f033dd-d550-4d42-b9f6-6dbfd0b57d70",
   "metadata": {},
   "source": [
    "## Run orthofinder 2.5.5 on proteomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a63b2868-7bad-4da2-9a16-5cab8be16ae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "orthofinder_output_dir = Path(\"results/1_orthofinder_analysis/data/orthofinder_results\")\n",
    "append = \"Acinetobacter\"\n",
    "#ru.run_orthofinder(str(proteomes_dir), o=orthofinder_output_dir, n=append)\n",
    "orthofinder_output_dir = orthofinder_output_dir / f'Results_{append}'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad42f19e-b882-4c05-adbf-477549b544a2",
   "metadata": {},
   "source": [
    "### Parse orthofinder output to find HOGs. \n",
    "- The below code will may not be stable upon re-running OrthoFinder, HOGs will likely have different names etc."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "b16a9744-ddf8-435c-a848-651bba4a5de7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#make summary files\n",
    "summary_dir = orthofinder_output_dir / 'summary_data'\n",
    "proteomes_dir = orthofinder_output_dir / 'proteomes'\n",
    "gffs_dir = Path(orthofinder_output_dir / 'gffs')\n",
    "\n",
    "summary_dir.mkdir(exist_ok=True)\n",
    "proteomes_dir.mkdir(exist_ok=True)\n",
    "gffs_dir.mkdir(exist_ok=True, parents=True)\n",
    "\n",
    "summary_df['organism_name'].to_json(summary_dir / 'AssemblyAccession_to_SpeciesName.json')\n",
    "\n",
    "## move gffs to folder\n",
    "nu.move_gffs(    data_dir  = './ncbi_dataset/ncbi_dataset/data/', new_gff_dir = gffs_dir)\n",
    "nu.move_proteomes('./ncbi_dataset/ncbi_dataset/data/', proteomes_dir)\n",
    "dop_obj = dop.DashOrthoParser(orthofinder_output_dir)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "aac96cd9-bf2d-45aa-b596-a3677c1ebdff",
   "metadata": {},
   "outputs": [],
   "source": [
    "dop_obj.accession_to_name\n",
    "'GCF_001077675.1'\n",
    "genome_annotation_df = dop_obj.make_genome_annotation_df('GCF_001077675.1',get_common_names=True)\n",
    "astA = 'gene-ACX60_RS01825'\n",
    "\n",
    "astR = 'gene-ACX60_RS12905'\n",
    "astN = 'gene-ACX60_RS12900'\n",
    "astO = 'gene-ACX60_RS12895'\n",
    "astP = 'gene-ACX60_RS12890'\n",
    "\n",
    "astR_HOG = genome_annotation_df.set_index('Parents').loc['gene-ACX60_RS12905', 'HOGs']\n",
    "astR_orthologs = dop_obj.all_prots_in_HOG(astR_HOG)\n",
    "\n",
    "astR_accs = [a[:15][::-1].replace('_', '.', 1)[::-1] for a in astR_orthologs]\n",
    "astR_proteins = dop_obj.get_HOG_protein_seqs(astR_HOG)\n",
    "astR_seqs_d = astR_proteins['Protein_seq'].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "1b712645-c3a8-4b31-b2f3-113639b55c14",
   "metadata": {},
   "outputs": [],
   "source": [
    "tree_dir = Path('./results/1_orthofinder_analysis/raxML_tree')\n",
    "tree_dir.mkdir(exist_ok=True)\n",
    "astR_seqs_fp = tree_dir / 'astR_orthologs.faa'\n",
    "pfa.write_to_fasta(astR_seqs_d, astR_seqs_fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "c2cf646c-7505-41cb-8995-32f88569df0f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: muscle -in /var/folders/vw/7lg51dfd3ql9g_j55xz_3dqh0000gn/T/muscle_1htbnm_u.fa -out results/1_orthofinder_analysis/raxML_tree/astR_orthologs.muscle.aln\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "muscle_1htbnm_u 12 seqs, lengths min 140, max 153, avg 141\n",
      "00:00:00      2 MB(0%)  Iter   1    1.28%  K-mer dist pass 1\n",
      "00:00:00      2 MB(0%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00      2 MB(0%)  Iter   1    1.28%  K-mer dist pass 2\n",
      "00:00:00      2 MB(0%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00      2 MB(0%)  Iter   1    9.09%  Align node       \n",
      "00:00:00      3 MB(0%)  Iter   1   18.18%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   27.27%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   36.36%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   45.45%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   54.55%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   63.64%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   72.73%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   81.82%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1   90.91%  Align node\n",
      "00:00:00      3 MB(0%)  Iter   1  100.00%  Align node\n",
      "00:00:00      4 MB(0%)  Iter   1  100.00%  Align node\n",
      "00:00:00      4 MB(0%)  Iter   1    8.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   16.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   25.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   33.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   41.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   50.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   58.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   66.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   75.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   83.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1   91.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   10.00%  Refine tree   \n",
      "00:00:00      4 MB(0%)  Iter   2   20.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   30.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   40.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   50.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   60.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   70.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2   80.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2  100.00%  Refine tree\n",
      "00:00:00      4 MB(0%)  Iter   2    8.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   16.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   25.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   33.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   41.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   50.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   58.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   66.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   75.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   83.33%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2   91.67%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00      4 MB(0%)  Iter   3    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   3  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   4  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   5  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   6  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   7  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8    9.52%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   14.29%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   19.05%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   23.81%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   28.57%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   33.33%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   38.10%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   42.86%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   47.62%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   52.38%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   57.14%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   61.90%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   66.67%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   71.43%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   76.19%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   80.95%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   85.71%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   90.48%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8   95.24%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8  100.00%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8  104.76%  Refine biparts\n",
      "00:00:00      4 MB(0%)  Iter   8  100.00%  Refine biparts\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'results/1_orthofinder_analysis/raxML_tree/astR_orthologs.muscle.aln'"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "muscle_out= str(astR_seqs_fp).replace('.faa', '.muscle.aln')\n",
    "importlib.reload(ef)\n",
    "ef.run_muscle(astR_seqs_d, muscle_out)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "995910ae-5167-44f0-a4ac-c740993d691e",
   "metadata": {},
   "source": [
    "### Load alignment and Run RAxML"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "68ad726b-4a85-4415-8b75-92253760f3da",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running: raxmlHPC -T 8 -m PROTGAMMALG -p 12345 -x 12345 -# 100 -f a -s /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/astR_orthologs.muscle.aln -n AstR -w /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output\n",
      "Option -T does not have any effect with the sequential or parallel MPI version.\n",
      "It is used to specify the number of threads for the Pthreads-based parallelization\n",
      "Warning, you specified a working directory via \"-w\"\n",
      "Keep in mind that RAxML only accepts absolute path names, not relative ones!\n",
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "\n",
      "IMPORTANT WARNING: Sequences GCF_000021245_2_WP_000368712.1 and GCF_000770605_1_WP_000368712.1 are exactly identical\n",
      "\n",
      "IMPORTANT WARNING\n",
      "Found 1 sequence that is exactly identical to other sequences in the alignment.\n",
      "Normally they should be excluded from the analysis.\n",
      "\n",
      "Just in case you might need it, an alignment file with \n",
      "sequence duplicates removed is printed to file /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/astR_orthologs.muscle.aln.reduced\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 129 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 8.39%\n",
      "\n",
      "RAxML rapid bootstrapping and subsequent ML search\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "\n",
      "Executing 100 rapid bootstrap inferences and thereafter a thorough ML search \n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "GAMMA model of rate heterogeneity, ML estimate of alpha-parameter\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 129\n",
      "Name: No Name Provided\n",
      "DataType: AA\n",
      "Substitution Matrix: LG\n",
      "Using fixed base frequencies\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "raxmlHPC -T 8 -m PROTGAMMALG -p 12345 -x 12345 -# 100 -f a -s /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/astR_orthologs.muscle.aln -n AstR -w /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output \n",
      "\n",
      "\n",
      "\n",
      "Time for BS model parameter optimization 0.022330\n",
      "Bootstrap[0]: Time 0.370475 seconds, bootstrap likelihood -1298.382679, best rearrangement setting 13\n",
      "Bootstrap[1]: Time 0.234043 seconds, bootstrap likelihood -1136.255044, best rearrangement setting 13\n",
      "Bootstrap[2]: Time 0.184860 seconds, bootstrap likelihood -1186.937172, best rearrangement setting 12\n",
      "Bootstrap[3]: Time 0.270794 seconds, bootstrap likelihood -1218.103327, best rearrangement setting 15\n",
      "Bootstrap[4]: Time 0.086233 seconds, bootstrap likelihood -1172.390963, best rearrangement setting 10\n",
      "Bootstrap[5]: Time 0.200405 seconds, bootstrap likelihood -1246.939279, best rearrangement setting 10\n",
      "Bootstrap[6]: Time 0.191902 seconds, bootstrap likelihood -1084.197634, best rearrangement setting 6\n",
      "Bootstrap[7]: Time 0.072871 seconds, bootstrap likelihood -1243.283855, best rearrangement setting 14\n",
      "Bootstrap[8]: Time 0.084783 seconds, bootstrap likelihood -1152.726292, best rearrangement setting 5\n",
      "Bootstrap[9]: Time 0.239121 seconds, bootstrap likelihood -1264.841074, best rearrangement setting 7\n",
      "Bootstrap[10]: Time 0.298244 seconds, bootstrap likelihood -1213.006728, best rearrangement setting 7\n",
      "Bootstrap[11]: Time 0.237703 seconds, bootstrap likelihood -1269.609336, best rearrangement setting 5\n",
      "Bootstrap[12]: Time 0.233151 seconds, bootstrap likelihood -1251.838798, best rearrangement setting 8\n",
      "Bootstrap[13]: Time 0.253154 seconds, bootstrap likelihood -1207.963703, best rearrangement setting 7\n",
      "Bootstrap[14]: Time 0.227350 seconds, bootstrap likelihood -1175.003210, best rearrangement setting 12\n",
      "Bootstrap[15]: Time 0.088534 seconds, bootstrap likelihood -1205.945660, best rearrangement setting 15\n",
      "Bootstrap[16]: Time 0.255553 seconds, bootstrap likelihood -1216.047811, best rearrangement setting 12\n",
      "Bootstrap[17]: Time 0.242834 seconds, bootstrap likelihood -1237.019677, best rearrangement setting 12\n",
      "Bootstrap[18]: Time 0.272281 seconds, bootstrap likelihood -1213.907225, best rearrangement setting 5\n",
      "Bootstrap[19]: Time 0.112084 seconds, bootstrap likelihood -1243.670773, best rearrangement setting 8\n",
      "Bootstrap[20]: Time 0.092933 seconds, bootstrap likelihood -1262.689117, best rearrangement setting 10\n",
      "Bootstrap[21]: Time 0.228838 seconds, bootstrap likelihood -1285.593673, best rearrangement setting 6\n",
      "Bootstrap[22]: Time 0.298437 seconds, bootstrap likelihood -1198.916904, best rearrangement setting 5\n",
      "Bootstrap[23]: Time 0.197548 seconds, bootstrap likelihood -1095.033306, best rearrangement setting 7\n",
      "Bootstrap[24]: Time 0.300830 seconds, bootstrap likelihood -1206.005096, best rearrangement setting 8\n",
      "Bootstrap[25]: Time 0.208099 seconds, bootstrap likelihood -1144.814633, best rearrangement setting 6\n",
      "Bootstrap[26]: Time 0.235065 seconds, bootstrap likelihood -1184.830121, best rearrangement setting 12\n",
      "Bootstrap[27]: Time 0.215400 seconds, bootstrap likelihood -1141.466529, best rearrangement setting 10\n",
      "Bootstrap[28]: Time 0.263732 seconds, bootstrap likelihood -1122.852067, best rearrangement setting 11\n",
      "Bootstrap[29]: Time 0.245691 seconds, bootstrap likelihood -1167.643028, best rearrangement setting 12\n",
      "Bootstrap[30]: Time 0.213761 seconds, bootstrap likelihood -1232.103879, best rearrangement setting 15\n",
      "Bootstrap[31]: Time 0.314777 seconds, bootstrap likelihood -1237.605594, best rearrangement setting 15\n",
      "Bootstrap[32]: Time 0.319039 seconds, bootstrap likelihood -1179.392366, best rearrangement setting 6\n",
      "Bootstrap[33]: Time 0.231854 seconds, bootstrap likelihood -1218.959615, best rearrangement setting 15\n",
      "Bootstrap[34]: Time 0.257696 seconds, bootstrap likelihood -1176.224801, best rearrangement setting 11\n",
      "Bootstrap[35]: Time 0.210330 seconds, bootstrap likelihood -1168.267387, best rearrangement setting 8\n",
      "Bootstrap[36]: Time 0.270154 seconds, bootstrap likelihood -1240.226414, best rearrangement setting 9\n",
      "Bootstrap[37]: Time 0.213853 seconds, bootstrap likelihood -1218.623968, best rearrangement setting 11\n",
      "Bootstrap[38]: Time 0.281483 seconds, bootstrap likelihood -1209.848108, best rearrangement setting 8\n",
      "Bootstrap[39]: Time 0.282459 seconds, bootstrap likelihood -1304.075004, best rearrangement setting 13\n",
      "Bootstrap[40]: Time 0.098132 seconds, bootstrap likelihood -1161.335121, best rearrangement setting 13\n",
      "Bootstrap[41]: Time 0.103095 seconds, bootstrap likelihood -1278.809163, best rearrangement setting 10\n",
      "Bootstrap[42]: Time 0.089904 seconds, bootstrap likelihood -1268.219698, best rearrangement setting 5\n",
      "Bootstrap[43]: Time 0.434671 seconds, bootstrap likelihood -1206.232167, best rearrangement setting 14\n",
      "Bootstrap[44]: Time 0.266653 seconds, bootstrap likelihood -1244.172568, best rearrangement setting 10\n",
      "Bootstrap[45]: Time 0.342778 seconds, bootstrap likelihood -1156.446815, best rearrangement setting 15\n",
      "Bootstrap[46]: Time 0.257530 seconds, bootstrap likelihood -1188.944552, best rearrangement setting 14\n",
      "Bootstrap[47]: Time 0.288598 seconds, bootstrap likelihood -1194.483958, best rearrangement setting 8\n",
      "Bootstrap[48]: Time 0.244863 seconds, bootstrap likelihood -1259.005340, best rearrangement setting 6\n",
      "Bootstrap[49]: Time 0.291918 seconds, bootstrap likelihood -1304.955823, best rearrangement setting 7\n",
      "Bootstrap[50]: Time 0.290969 seconds, bootstrap likelihood -1214.597681, best rearrangement setting 7\n",
      "Bootstrap[51]: Time 0.244232 seconds, bootstrap likelihood -1107.794493, best rearrangement setting 11\n",
      "Bootstrap[52]: Time 0.210129 seconds, bootstrap likelihood -1147.775160, best rearrangement setting 12\n",
      "Bootstrap[53]: Time 0.259081 seconds, bootstrap likelihood -1241.633907, best rearrangement setting 8\n",
      "Bootstrap[54]: Time 0.245309 seconds, bootstrap likelihood -1210.495860, best rearrangement setting 9\n",
      "Bootstrap[55]: Time 0.199360 seconds, bootstrap likelihood -1272.641536, best rearrangement setting 11\n",
      "Bootstrap[56]: Time 0.314130 seconds, bootstrap likelihood -1219.394378, best rearrangement setting 5\n",
      "Bootstrap[57]: Time 0.275958 seconds, bootstrap likelihood -1203.563276, best rearrangement setting 7\n",
      "Bootstrap[58]: Time 0.085632 seconds, bootstrap likelihood -1188.735211, best rearrangement setting 10\n",
      "Bootstrap[59]: Time 0.315550 seconds, bootstrap likelihood -1247.957881, best rearrangement setting 9\n",
      "Bootstrap[60]: Time 0.206402 seconds, bootstrap likelihood -1154.982927, best rearrangement setting 7\n",
      "Bootstrap[61]: Time 0.289706 seconds, bootstrap likelihood -1141.138643, best rearrangement setting 5\n",
      "Bootstrap[62]: Time 0.266539 seconds, bootstrap likelihood -1247.737614, best rearrangement setting 11\n",
      "Bootstrap[63]: Time 0.269365 seconds, bootstrap likelihood -1167.090565, best rearrangement setting 12\n",
      "Bootstrap[64]: Time 0.232805 seconds, bootstrap likelihood -1191.589385, best rearrangement setting 7\n",
      "Bootstrap[65]: Time 0.201221 seconds, bootstrap likelihood -1129.850032, best rearrangement setting 5\n",
      "Bootstrap[66]: Time 0.287015 seconds, bootstrap likelihood -1232.039900, best rearrangement setting 5\n",
      "Bootstrap[67]: Time 0.256628 seconds, bootstrap likelihood -1237.561569, best rearrangement setting 15\n",
      "Bootstrap[68]: Time 0.095509 seconds, bootstrap likelihood -1272.077149, best rearrangement setting 11\n",
      "Bootstrap[69]: Time 0.254886 seconds, bootstrap likelihood -1248.419145, best rearrangement setting 5\n",
      "Bootstrap[70]: Time 0.163149 seconds, bootstrap likelihood -1265.707988, best rearrangement setting 8\n",
      "Bootstrap[71]: Time 0.222917 seconds, bootstrap likelihood -1171.946151, best rearrangement setting 10\n",
      "Bootstrap[72]: Time 0.269413 seconds, bootstrap likelihood -1339.591340, best rearrangement setting 11\n",
      "Bootstrap[73]: Time 0.303771 seconds, bootstrap likelihood -1197.781481, best rearrangement setting 9\n",
      "Bootstrap[74]: Time 0.192028 seconds, bootstrap likelihood -1265.986708, best rearrangement setting 5\n",
      "Bootstrap[75]: Time 0.082882 seconds, bootstrap likelihood -1235.793500, best rearrangement setting 8\n",
      "Bootstrap[76]: Time 0.080356 seconds, bootstrap likelihood -1132.492313, best rearrangement setting 9\n",
      "Bootstrap[77]: Time 0.249708 seconds, bootstrap likelihood -1368.651281, best rearrangement setting 11\n",
      "Bootstrap[78]: Time 0.288149 seconds, bootstrap likelihood -1227.226643, best rearrangement setting 15\n",
      "Bootstrap[79]: Time 0.326831 seconds, bootstrap likelihood -1220.978291, best rearrangement setting 9\n",
      "Bootstrap[80]: Time 0.230238 seconds, bootstrap likelihood -1245.348832, best rearrangement setting 15\n",
      "Bootstrap[81]: Time 0.230123 seconds, bootstrap likelihood -1133.390160, best rearrangement setting 7\n",
      "Bootstrap[82]: Time 0.233178 seconds, bootstrap likelihood -1302.759302, best rearrangement setting 6\n",
      "Bootstrap[83]: Time 0.171216 seconds, bootstrap likelihood -1209.654914, best rearrangement setting 10\n",
      "Bootstrap[84]: Time 0.190476 seconds, bootstrap likelihood -1316.400192, best rearrangement setting 11\n",
      "Bootstrap[85]: Time 0.091323 seconds, bootstrap likelihood -1235.884820, best rearrangement setting 8\n",
      "Bootstrap[86]: Time 0.203863 seconds, bootstrap likelihood -1146.134796, best rearrangement setting 7\n",
      "Bootstrap[87]: Time 0.257112 seconds, bootstrap likelihood -1196.130594, best rearrangement setting 7\n",
      "Bootstrap[88]: Time 0.230602 seconds, bootstrap likelihood -1215.003426, best rearrangement setting 12\n",
      "Bootstrap[89]: Time 0.204707 seconds, bootstrap likelihood -1189.350790, best rearrangement setting 10\n",
      "Bootstrap[90]: Time 0.235743 seconds, bootstrap likelihood -1250.212728, best rearrangement setting 8\n",
      "Bootstrap[91]: Time 0.289199 seconds, bootstrap likelihood -1180.351151, best rearrangement setting 12\n",
      "Bootstrap[92]: Time 0.292485 seconds, bootstrap likelihood -1125.003079, best rearrangement setting 13\n",
      "Bootstrap[93]: Time 0.270866 seconds, bootstrap likelihood -1197.087263, best rearrangement setting 15\n",
      "Bootstrap[94]: Time 0.225446 seconds, bootstrap likelihood -1168.527557, best rearrangement setting 5\n",
      "Bootstrap[95]: Time 0.296266 seconds, bootstrap likelihood -1260.842001, best rearrangement setting 5\n",
      "Bootstrap[96]: Time 0.246409 seconds, bootstrap likelihood -1186.546168, best rearrangement setting 13\n",
      "Bootstrap[97]: Time 0.246325 seconds, bootstrap likelihood -1117.124440, best rearrangement setting 11\n",
      "Bootstrap[98]: Time 0.215395 seconds, bootstrap likelihood -1214.974119, best rearrangement setting 6\n",
      "Bootstrap[99]: Time 0.307385 seconds, bootstrap likelihood -1208.886177, best rearrangement setting 14\n",
      "\n",
      "\n",
      "Overall Time for 100 Rapid Bootstraps 23.114079 seconds\n",
      "Average Time per Rapid Bootstrap 0.231141 seconds\n",
      "\n",
      "Starting ML Search ...\n",
      "\n",
      "Fast ML optimization finished\n",
      "\n",
      "Fast ML search Time: 17.228134 seconds\n",
      "\n",
      "Slow ML Search 0 Likelihood: -1229.847513\n",
      "Slow ML Search 1 Likelihood: -1229.847513\n",
      "Slow ML Search 2 Likelihood: -1229.847513\n",
      "Slow ML Search 3 Likelihood: -1229.847513\n",
      "Slow ML Search 4 Likelihood: -1229.847513\n",
      "Slow ML Search 5 Likelihood: -1229.847513\n",
      "Slow ML Search 6 Likelihood: -1229.847513\n",
      "Slow ML Search 7 Likelihood: -1229.847513\n",
      "Slow ML Search 8 Likelihood: -1229.847513\n",
      "Slow ML Search 9 Likelihood: -1229.847513\n",
      "Slow ML optimization finished\n",
      "\n",
      "Slow ML search Time: 5.290530 seconds\n",
      "Thorough ML search Time: 0.467573 seconds\n",
      "\n",
      "Final ML Optimization Likelihood: -1229.847513\n",
      "\n",
      "Model Information:\n",
      "\n",
      "Model Parameters of Partition 0, Name: No Name Provided, Type of Data: AA\n",
      "alpha: 1.935297\n",
      "Tree-Length: 2.551748\n",
      "rate A <-> R: 0.399182\n",
      "rate A <-> N: 0.259945\n",
      "rate A <-> D: 0.371058\n",
      "rate A <-> C: 2.337364\n",
      "rate A <-> Q: 0.910775\n",
      "rate A <-> E: 0.975241\n",
      "rate A <-> G: 1.940106\n",
      "rate A <-> H: 0.336984\n",
      "rate A <-> I: 0.140697\n",
      "rate A <-> L: 0.371240\n",
      "rate A <-> K: 0.503815\n",
      "rate A <-> M: 1.055520\n",
      "rate A <-> F: 0.238237\n",
      "rate A <-> P: 1.105868\n",
      "rate A <-> S: 4.439041\n",
      "rate A <-> T: 2.009090\n",
      "rate A <-> W: 0.169702\n",
      "rate A <-> Y: 0.205613\n",
      "rate A <-> V: 2.392567\n",
      "rate R <-> N: 0.706048\n",
      "rate R <-> D: 0.116398\n",
      "rate R <-> C: 0.501968\n",
      "rate R <-> Q: 2.636754\n",
      "rate R <-> E: 0.341785\n",
      "rate R <-> G: 0.366408\n",
      "rate R <-> H: 2.278690\n",
      "rate R <-> I: 0.119250\n",
      "rate R <-> L: 0.283449\n",
      "rate R <-> K: 5.940467\n",
      "rate R <-> M: 0.454623\n",
      "rate R <-> F: 0.049508\n",
      "rate R <-> P: 0.312264\n",
      "rate R <-> S: 0.805843\n",
      "rate R <-> T: 0.543695\n",
      "rate R <-> W: 0.557424\n",
      "rate R <-> Y: 0.295274\n",
      "rate R <-> V: 0.160471\n",
      "rate N <-> D: 4.766737\n",
      "rate N <-> C: 0.496537\n",
      "rate N <-> Q: 1.592389\n",
      "rate N <-> E: 0.508692\n",
      "rate N <-> G: 1.350015\n",
      "rate N <-> H: 4.234381\n",
      "rate N <-> I: 0.179830\n",
      "rate N <-> L: 0.064256\n",
      "rate N <-> K: 2.014327\n",
      "rate N <-> M: 0.348390\n",
      "rate N <-> F: 0.084068\n",
      "rate N <-> P: 0.151925\n",
      "rate N <-> S: 3.764032\n",
      "rate N <-> T: 1.878729\n",
      "rate N <-> W: 0.042610\n",
      "rate N <-> Y: 0.574720\n",
      "rate N <-> V: 0.078587\n",
      "rate D <-> C: 0.058743\n",
      "rate D <-> Q: 0.491483\n",
      "rate D <-> E: 4.924234\n",
      "rate D <-> G: 0.793424\n",
      "rate D <-> H: 0.870603\n",
      "rate D <-> I: 0.010038\n",
      "rate D <-> L: 0.014157\n",
      "rate D <-> K: 0.265711\n",
      "rate D <-> M: 0.023991\n",
      "rate D <-> F: 0.016354\n",
      "rate D <-> P: 0.370412\n",
      "rate D <-> S: 1.164675\n",
      "rate D <-> T: 0.399902\n",
      "rate D <-> W: 0.028068\n",
      "rate D <-> Y: 0.126872\n",
      "rate D <-> V: 0.035653\n",
      "rate C <-> Q: 0.079639\n",
      "rate C <-> E: 0.003286\n",
      "rate C <-> G: 0.534566\n",
      "rate C <-> H: 0.601499\n",
      "rate C <-> I: 0.301083\n",
      "rate C <-> L: 0.557800\n",
      "rate C <-> K: 0.012457\n",
      "rate C <-> M: 0.839207\n",
      "rate C <-> F: 1.037881\n",
      "rate C <-> P: 0.070787\n",
      "rate C <-> S: 2.614753\n",
      "rate C <-> T: 1.073780\n",
      "rate C <-> W: 0.629281\n",
      "rate C <-> Y: 1.094488\n",
      "rate C <-> V: 1.839864\n",
      "rate Q <-> E: 3.876936\n",
      "rate Q <-> G: 0.251626\n",
      "rate Q <-> H: 4.520102\n",
      "rate Q <-> I: 0.068413\n",
      "rate Q <-> L: 0.546954\n",
      "rate Q <-> K: 3.037150\n",
      "rate Q <-> M: 1.570619\n",
      "rate Q <-> F: 0.033669\n",
      "rate Q <-> P: 0.586241\n",
      "rate Q <-> S: 1.149231\n",
      "rate Q <-> T: 1.014297\n",
      "rate Q <-> W: 0.221802\n",
      "rate Q <-> Y: 0.241650\n",
      "rate Q <-> V: 0.197511\n",
      "rate E <-> G: 0.327583\n",
      "rate E <-> H: 0.398044\n",
      "rate E <-> I: 0.041567\n",
      "rate E <-> L: 0.065426\n",
      "rate E <-> K: 1.697022\n",
      "rate E <-> M: 0.163145\n",
      "rate E <-> F: 0.017664\n",
      "rate E <-> P: 0.393844\n",
      "rate E <-> S: 0.574671\n",
      "rate E <-> T: 0.567695\n",
      "rate E <-> W: 0.073107\n",
      "rate E <-> Y: 0.112720\n",
      "rate E <-> V: 0.230098\n",
      "rate G <-> H: 0.292498\n",
      "rate G <-> I: 0.008174\n",
      "rate G <-> L: 0.041563\n",
      "rate G <-> K: 0.278555\n",
      "rate G <-> M: 0.131033\n",
      "rate G <-> F: 0.084125\n",
      "rate G <-> P: 0.184955\n",
      "rate G <-> S: 1.633930\n",
      "rate G <-> T: 0.121922\n",
      "rate G <-> W: 0.252125\n",
      "rate G <-> Y: 0.051346\n",
      "rate G <-> V: 0.072026\n",
      "rate H <-> I: 0.102245\n",
      "rate H <-> L: 0.343988\n",
      "rate H <-> K: 0.654763\n",
      "rate H <-> M: 0.415502\n",
      "rate H <-> F: 0.640560\n",
      "rate H <-> P: 0.477834\n",
      "rate H <-> S: 0.929667\n",
      "rate H <-> T: 0.548649\n",
      "rate H <-> W: 0.560661\n",
      "rate H <-> Y: 4.983361\n",
      "rate H <-> V: 0.111759\n",
      "rate I <-> L: 3.892408\n",
      "rate I <-> K: 0.149373\n",
      "rate I <-> M: 4.013113\n",
      "rate I <-> F: 1.044902\n",
      "rate I <-> P: 0.073509\n",
      "rate I <-> S: 0.060198\n",
      "rate I <-> T: 0.970728\n",
      "rate I <-> W: 0.104854\n",
      "rate I <-> Y: 0.218350\n",
      "rate I <-> V: 10.000000\n",
      "rate L <-> K: 0.129119\n",
      "rate L <-> M: 5.927594\n",
      "rate L <-> F: 2.434657\n",
      "rate L <-> P: 0.233879\n",
      "rate L <-> S: 0.171176\n",
      "rate L <-> T: 0.284471\n",
      "rate L <-> W: 0.581863\n",
      "rate L <-> Y: 0.281383\n",
      "rate L <-> V: 1.598956\n",
      "rate K <-> M: 0.616581\n",
      "rate K <-> F: 0.022460\n",
      "rate K <-> P: 0.366530\n",
      "rate K <-> S: 0.703048\n",
      "rate K <-> T: 1.067567\n",
      "rate K <-> W: 0.046864\n",
      "rate K <-> Y: 0.123890\n",
      "rate K <-> V: 0.173913\n",
      "rate M <-> F: 1.689205\n",
      "rate M <-> P: 0.093763\n",
      "rate M <-> S: 0.325811\n",
      "rate M <-> T: 1.897216\n",
      "rate M <-> W: 0.653740\n",
      "rate M <-> Y: 0.451968\n",
      "rate M <-> V: 1.782983\n",
      "rate F <-> P: 0.088706\n",
      "rate F <-> S: 0.339765\n",
      "rate F <-> T: 0.154944\n",
      "rate F <-> W: 2.307349\n",
      "rate F <-> Y: 7.328222\n",
      "rate F <-> V: 0.614777\n",
      "rate P <-> S: 1.256567\n",
      "rate P <-> T: 0.536635\n",
      "rate P <-> W: 0.089332\n",
      "rate P <-> Y: 0.084151\n",
      "rate P <-> V: 0.278428\n",
      "rate S <-> T: 6.077767\n",
      "rate S <-> W: 0.233693\n",
      "rate S <-> Y: 0.376132\n",
      "rate S <-> V: 0.092373\n",
      "rate T <-> W: 0.132241\n",
      "rate T <-> Y: 0.230856\n",
      "rate T <-> V: 2.054781\n",
      "rate W <-> Y: 2.959699\n",
      "rate W <-> V: 0.177959\n",
      "rate Y <-> V: 0.234116\n",
      "\n",
      "freq pi(A): 0.079066\n",
      "freq pi(R): 0.055941\n",
      "freq pi(N): 0.041977\n",
      "freq pi(D): 0.053052\n",
      "freq pi(C): 0.012937\n",
      "freq pi(Q): 0.040767\n",
      "freq pi(E): 0.071586\n",
      "freq pi(G): 0.057337\n",
      "freq pi(H): 0.022355\n",
      "freq pi(I): 0.062157\n",
      "freq pi(L): 0.099081\n",
      "freq pi(K): 0.064600\n",
      "freq pi(M): 0.022951\n",
      "freq pi(F): 0.042302\n",
      "freq pi(P): 0.044040\n",
      "freq pi(S): 0.061197\n",
      "freq pi(T): 0.053287\n",
      "freq pi(W): 0.012066\n",
      "freq pi(Y): 0.034155\n",
      "freq pi(V): 0.069146\n",
      "\n",
      "\n",
      "ML search took 22.991832 secs or 0.006387 hours\n",
      "\n",
      "Combined Bootstrap and ML search took 46.105990 secs or 0.012807 hours\n",
      "\n",
      "Drawing Bootstrap Support Values on best-scoring ML tree ...\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bestTree.AstR\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bestTree.AstR\n",
      "\n",
      "Program execution info written to /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_info.AstR\n",
      "All 100 bootstrapped trees written to: /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bootstrap.AstR\n",
      "\n",
      "Best-scoring ML tree written to: /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bestTree.AstR\n",
      "\n",
      "Best-scoring ML tree with support values written to: /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bipartitions.AstR\n",
      "\n",
      "Best-scoring ML tree with support values as branch labels written to: /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output/RAxML_bipartitionsBranchLabels.AstR\n",
      "\n",
      "Overall execution time for full ML analysis: 46.107448 secs or 0.012808 hours or 0.000534 days\n",
      "\n",
      "RAxML finished. Outputs in /Users/jonathanwinkelman/Trestle/Palmer_lab/Geary_et_al_2025/results/1_orthofinder_analysis/raxML_tree/raxML_output\n"
     ]
    }
   ],
   "source": [
    "raxml_out_dir = tree_dir / 'raxML_output'\n",
    "importlib.reload(ef)\n",
    "ef.run_raxml(muscle_out, raxml_out_dir, prefix='AstR', threads=8, model='LG', raxml_bin='raxmlHPC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "ffd5a46a-8f22-4908-84db-49c55072fef5",
   "metadata": {},
   "outputs": [],
   "source": [
    "itol_annotations_dir = tree_dir / 'itol_annotations'\n",
    "itol_annotations_dir.mkdir(exist_ok=True)\n",
    "tree = Phylo.read(file=f'{raxml_out_dir}/RAxML_bipartitionsBranchLabels.AstR', format='newick')\n",
    "relable_d = {}\n",
    "for cl in tree.get_terminals():\n",
    "    acc = cl.name[:15][::-1].replace('_', '.', 1)[::-1]\n",
    "    relable_d[cl.name] = dop_obj.accession_to_name[acc]\n",
    "itol_relable_out = itol_annotations_dir / 'RELABLE_RAxML_bipartitionsBranchLabels.astR'\n",
    "ia.relabel_itol_treeleafs(tree,relable_d, itol_relable_out )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "21005910-91f7-450e-ad2d-e4b9b74e535d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8298dfe-7558-4558-96b6-7cb505647216",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ee953c9-6e0f-449b-a5b9-496de3b9ed7e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87f592f5-4a5c-4e5d-8558-d9236e6b8362",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "22382523-48dc-4462-a66a-8d9c57e1dbca",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c6f9077-b45a-4a3a-b0c6-eb3c28ec1ba3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60fc8d7a-cf70-478d-b05f-0050d06f9770",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f589e5a7-0682-43cd-8d6e-1c92d10a514a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48d2db64-1da9-4681-91da-cf9b8e91b262",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
